features/base/participants/reducer.js

```diff
@@ -73,45 +73,27 @@ function participant(state, action) {
         }
         return state;
 
-    /**
-     * Cleans conference-specific properties of the local participant when
-     * conference is left.
-     */
-    case CONFERENCE_LEFT:
-        if (state.local) {
-            return {
-                ...state,
-                id: LOCAL_PARTICIPANT_DEFAULT_ID,
-                avatar: state.avatar ||
-                    _getAvatarURL(LOCAL_PARTICIPANT_DEFAULT_ID, state.email),
-                focused: false,
-                pinned: false,
-                role: PARTICIPANT_ROLE.NONE,
-                selected: false,
-                speaking: false
-            };
-        }
-        return state;
-
-    case DOMINANT_SPEAKER_CHANGED:
-        // Only one dominant speaker is allowed.
-        return Object.assign({}, state, {
-            speaking: state.id === action.participant.id
-        });
-
     case PARTICIPANT_JOINED:
+        let participant = action.participant;
+        let participantId = participant.id
+            || (participant.local && LOCAL_PARTICIPANT_DEFAULT_ID);
```

Comment:
I'm afraid I find the above condition unclear to me.

If participant.id is not falsy, then participantId is that non-falsy value. That looks OK to me.

Else if participant.local is not falsy, then participantId is LOCAL_PARTICIPANT_DEFAULT_ID. That looks OK to me.

But if participant.local is falsy, then participantId is that falsy value. Is that what you wanted to express?